pkg/api/validation/validation.go

```diff
@@ -1811,6 +1811,30 @@ func validateImagePullSecrets(imagePullSecrets []api.LocalObjectReference, fldPa
 	return allErrors
 }
 
+// validateAffinity checks if given affinities are valid
+func validateAffinity(affinity *api.Affinity, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if affinity != nil {
+		if affinity.NodeAffinity != nil {
+			na := affinity.NodeAffinity
```

Comment:
you can probably do the variable swap trick on the if line: 

if na := affinity.NodeAffinity; na != nil {